src/cpp/fastdds/domain/DomainParticipantFactory.cpp

```diff
@@ -228,6 +228,55 @@ DomainParticipant* DomainParticipantFactory::create_participant(
     return dom_part;
 }
 
+DomainParticipant* DomainParticipantFactory::create_participant(
```

Comment:
The `crate_participant` defined on line 188 and this overload here have 90% of the code in common. We should make one of them dependent on the other, or extract the common part to an internal method.